core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -84,6 +87,7 @@
   private boolean fullStreamDecompression;
   private DecompressorRegistry decompressorRegistry = DecompressorRegistry.getDefaultInstance();
   private CompressorRegistry compressorRegistry = CompressorRegistry.getDefaultInstance();
+  private final Set<Cork> corks = Sets.newConcurrentHashSet();
```

Comment:
Is it possible to use an `AtomicInteger` instead?

Reply:
Yes, i exchange of having `AtomicBoolean closed` flag inside each cork. Changed.

I think this is closely related to the discussion bellow (if we should have cork/uncork methods).